### PR TITLE
chore: cleanup verbose flag

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

Yes

**If relevant, did you update the documentation?**
Yes

**Summary**

remove verbose flag as `--stats verbose` is sufficient

**Does this PR introduce a breaking change?**
Yes

**Other information**
Ref - https://github.com/webpack/webpack-cli/issues/1167

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -235,22 +235,6 @@ module.exports = {
             group: BASIC_GROUP,
             description: 'NodeJS flags',
         },
-        {
-            name: 'stats',
-            usage: '--stats <value>',
-            type: [String, Boolean],
```
Comment (snitin315) [inline packages/webpack-cli/lib/utils/cli-flags.js:241]: Seems you accidently removed --stats too.

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js]: 😞

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 2]
```diff
@@ -13,20 +13,14 @@ class StatsGroup extends GroupHelper {
     }
 
     resolveOptions() {
-        if (this.args.verbose && this.args.stats) {
-            logger.warn('Conflict between "verbose" and "stats" options. Using verbose.');
-            this.opts.options.stats = 'verbose';
+        const { stats, json } = this.args;
+        if (stats && !StatsGroup.validOptions().includes(stats)) {
+            logger.warn(`'${stats}' is invalid value for stats. Using 'normal' option for stats`);
+            this.opts.options.stats = 'normal';
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/StatsGroup.js:19]: We should not validate this, webpack do it for us :smile: Can we fix it here? And add tests with failing?

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/StatsGroup.js:19]: yes need fixing, on it

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/StatsGroup.js]: Done @evilebottnawi 👍

File: test/stats/cli-flags/stats.test.js [hunk 3]
```diff
@@ -29,17 +29,10 @@ describe('stats flag', () => {
...
-        expect(stderr).toBeTruthy();
-        expect(stderr).toContain('Conflict between "verbose" and "stats" options');
-        expect(stdout).toBeTruthy();
-    });
-
     it('should warn when an unknown flag stats value is passed', () => {
         const { stderr, stdout } = run(__dirname, ['--stats', 'foo']);
         expect(stderr).toBeTruthy();
-        expect(stderr).toContain('invalid value for stats');
+        expect(stderr).toContain('WebpackOptionsValidationError');
```
Comment (evilebottnawi) [inline test/stats/cli-flags/stats.test.js:35]: Can we improve message testing here and we can merge it :+1:

Comment (anshumanv) [reply test/stats/cli-flags/stats.test.js]: done 👍

**Comment by evilebottnawi** [comment]
After merge can you investigate other places with validation?

**Comment by anshumanv** [comment]
> After merge can you investigate other places with validation?

Yes will investigate 👍

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 4]
```diff
@@ -1,32 +1,18 @@
...
-            if (this.args.verbose) {
-                this.opts.options.stats = 'verbose';
-            } else if (this.args.stats && !StatsGroup.validOptions().includes(this.args.stats)) {
-                logger.warn(`'${this.args.stats}' is invalid value for stats. Using 'normal' option for stats`);
-                this.opts.options.stats = 'normal';
-            } else {
-                this.opts.options.stats = this.args.stats;
-            }
+        const { stats, json } = this.args;
+        if (typeof stats !== undefined) {
```
Comment (snitin315) [inline packages/webpack-cli/lib/groups/StatsGroup.js:12]: Shouldn't it be like this?

```suggestion
        if (typeof stats !== 'undefined') {
```

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/StatsGroup.js:12]: nope not a string

**Comment by anshumanv** [comment]
> Apart from suggestion by Nitin, Looks good to me.

addressed

File: packages/webpack-cli/lib/groups/StatsGroup.js (hunk 4 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/groups/StatsGroup.js:12]: simpler would be to check `stats !== undefined`

**Pull request merged**